monkey/infection_monkey/post_breach/actions/communicate_as_new_user.py

```diff
@@ -31,88 +32,76 @@ def __init__(self):
-                commandline = "ping -c 2 google.com"
-                linux_cmds.extend([";", "sudo", "-u", username, commandline])
-                final_command = ' '.join(linux_cmds)
-                logger.debug("Trying to execute these commands: {}".format(final_command))
-                output = subprocess.check_output(final_command, stderr=subprocess.STDOUT, shell=True)
-                PostBreachTelem(self, (
-                    CREATED_PROCESS_AS_USER_LINUX_FORMAT.format(commandline, username, output[:150]), True)).send()
-                return
-            except subprocess.CalledProcessError as e:
-                PostBreachTelem(self, (e.output, False)).send()
-                return
-
-    def try_to_create_user_windows(self, username, password):
+            self.communicate_as_new_user_linux(username)
+
+    def communicate_as_new_user_linux(self, username):
         try:
-            windows_cmds = BackdoorUser.get_windows_commands_to_add_user(username, password, True)
-            logger.debug("Trying these commands: {}".format(str(windows_cmds)))
-            subprocess.check_output(windows_cmds, stderr=subprocess.STDOUT, shell=True)
-            return True
+            linux_cmds = BackdoorUser.get_linux_commands_to_add_user(username)
+            commandline = "ping -c 2 google.com"
+            linux_cmds.extend([";", "sudo", "-u", username, commandline])
+            final_command = ' '.join(linux_cmds)
+            logger.debug("Trying to execute these commands: {}".format(final_command))
+            output = subprocess.check_output(final_command, stderr=subprocess.STDOUT, shell=True)
+            PostBreachTelem(self, (
+                CREATED_PROCESS_AS_USER_LINUX_FORMAT.format(commandline, username, output[:150]), True)).send()
         except subprocess.CalledProcessError as e:
+            PostBreachTelem(self, (e.output, False)).send()
+
+    def communicate_as_new_user_windows(self, username):
+        # Importing these only on windows, as they won't exist on linux.
+        import win32con
+        import win32process
+        import win32api
+
+        try:
+            with NewUser(username, PASSWORD) as new_user:
+                # Using os.path is OK, as this is on windows for sure
+                ping_app_path = os.path.join(os.environ["WINDIR"], "system32", "PING.exe")
+                if not os.path.exists(ping_app_path):
+                    PostBreachTelem(self, ("{} not found.".format(ping_app_path), False)).send()
+                    return  # Can't continue without ping.
+
+                try:
+                    # Open process as that user:
+                    # https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-createprocessasusera
+                    commandline = "{} {} {} {}".format(ping_app_path, "google.com", "-n", "2")
+                    process_handle = win32process.CreateProcessAsUser(
```

Comment:
process_handle*s* or *process_info* but can't name it process_handle when it contains also a thread handle

Reply:
Renamed